### PR TITLE
Adds Text Search

### PR DESCRIPTION
**Pull request by samuelechu**
#244 
image

### CODE REVIEW

File: src/main/webapp/components/text-search/text-search.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return props.onChange(getTextQuery(text))
+  }
+
+  return (
+    <React.Fragment>
+      <Paper
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: 16,
```
Comment (willwill96) [inline src/main/webapp/components/text-search/text-search.tsx:56]: Should we standardize the padding on all of our query builders?

File: src/main/webapp/components/text-search/text-search.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    ],
+  },
+})
+
+export default (props: QueryBuilderProps) => {
+  const [text, setText] = React.useState(
+    props.query && props.query.filterTree
+      ? (props.query.filterTree.filters[0] as QueryFilter).value
+      : '*'
+  )
```
Comment (willwill96) [inline src/main/webapp/components/text-search/text-search.tsx:43]: :pencil2: I think getIn would be cleaner here

File: src/main/webapp/components/text-search/text-search.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          display: 'flex',
+          alignItems: 'center',
+          padding: 16,
+          boxSizing: 'border-box',
+        }}
+      >
+        <TextSearchQueryBuilder text={text} onChange={onChange} />
+      </Paper>
+
+      <Divider />
```
Comment (willwill96) [inline src/main/webapp/components/text-search/text-search.tsx:63]: :pencil2: should this be here?

File: src/main/webapp/components/query-builder/query-builder.tsx [hunk 4]
```diff
@@ -16,7 +16,7 @@ export type QueryBuilderProps = {
   attributeDefinitions?: AttributeDefinition[]
   onChange: (query: QueryType) => void
   query?: QueryType
-  addOptionsRef: HTMLDivElement
+  addOptionsRef?: HTMLDivElement
```
Comment (willwill96) [inline src/main/webapp/components/query-builder/query-builder.tsx:19]: What is this change for?

File: src/main/webapp/components/text-search/text-search.tsx (hunk 3 above)
Comment (samuelechu) [reply src/main/webapp/components/text-search/text-search.tsx:63]: oops

**Pull request merged**